subprojects/core/src/test/groovy/org/gradle/initialization/DefaultSettingsTest.groovy

```diff
@@ -191,6 +191,14 @@ class DefaultSettingsTest extends Specification {
         settings.dynamicProp == 'value'
     }
 
+    def 'can get and set dynamic properties on extension'() {
```

Comment:
This feature is probably worth some kind of int test coverage.

Reply:
@JLLeitschuh please add a test case in `SettingsDslIntegrationSpec` that exercise the `ExtensionAware` API for extensions and extra properties